### PR TITLE
support custom config for backing stores

### PR DESCRIPTION
**Pull request by guymguym**
### Explain the changes
1. support disabling chunk digest/encryption/compression.
2. support disabling md5/sha256 for etag.
3. support increasing max object part size.
4. support using erasure coding by default.

### Issues: Fixed #xxx / Gap #xxx
1. NA

### Testing Instructions:
1. NA

### CODE REVIEW

**Review by romayalon — APPROVED** [review]

File: src/server/object_services/object_server.js [hunk 1]
```diff
@@ -1591,6 +1591,44 @@ function check_md_conditions(conditions, obj) {
...
+   const md5_b64 = updates?.md5_b64 || entity.md5_b64;
+   if (md5_b64) return Buffer.from(md5_b64, 'base64').toString('hex');
+
+   const sha256_b64 = updates?.sha256_b64 || entity.sha256_b64;
+   if (sha256_b64) return 'sha256-' + Buffer.from(sha256_b64, 'base64').toString('hex');
+
+   const id = updates?._id || entity._id;
+   if (id) return 'id-' + id.toHexString();
+
+   return '';
```
Comment (guymguym) [inline src/server/object_services/object_server.js:1629]: @romayalon I wonder if I should throw here instead of silently "failing" ... WDYT?

Comment (romayalon) [reply src/server/object_services/object_server.js:1629]: Is there a situation where we get there? I couldn't find one and I thought that id is the default

Comment (guymguym) [reply src/server/object_services/object_server.js:1629]: right, we shouldn't get there, this is why I thought it might be better to throw

Comment (guymguym) [reply src/server/object_services/object_server.js:1629]: right, we shouldn't get there, this is why I thought it might be better to throw

**Pull request merged**